frontend/src/pages/groups/GroupCreatePage.tsx

```diff
@@ -0,0 +1,66 @@
+import { FunctionComponent } from 'react'
+import LayoutWithNav from '../../layouts/LayoutWithNav'
+import { Group, GroupInput } from '../../types/api-types'
+import { gql, useMutation } from '@apollo/client'
+import { useHistory } from 'react-router-dom'
+import GroupForm from './GroupForm'
+
+const ADD_GROUP = gql`
+  mutation Groups($input: GroupInput!) {
+    addGroup(input: $input) {
+      id
+    }
+  }
+`
+
+const GroupCreatePage: FunctionComponent = () => {
+  const history = useHistory()
+
+  const [
+    addGroup,
+    { loading: mutationIsLoading, error: mutationError },
+  ] = useMutation<{ addGroup: Group }>(ADD_GROUP)
+
+  const onSubmit = (input: GroupInput) => {
+    // Create the group and then redirect to its view/edit page
+    addGroup({ variables: { input } })
+      .then(({ data }) => {
+        if (data) {
+          const newGroupId = data.addGroup.id
+          history.push(`/group/${newGroupId}`)
```

Comment:
Thought (non-blocking): I think it's outside the scope of this PR, but it'd be nice to define routes somewhere common where the server and client could both read them so the server didn't have to guess/assume at what was a valid client-side route.

Reply:
Defining them that way would also eliminate the class of bugs where the routes in the react router mismatch with hand-typed routes like this one.

Reply:
I agree! I've opened #68 to take care of that.